src/sparse_blas/backends/mklgpu/mklgpu_spmm.cpp

```diff
@@ -0,0 +1,30 @@
+/***************************************************************************
+*  Copyright (C) Codeplay Software Limited
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+*  For your convenience, a copy of the License has been included in this
+*  repository.
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License.
+*
+**************************************************************************/
+
+#include "sparse_blas/backends/mkl_common/mkl_helper.hpp"
+#include "sparse_blas/macros.hpp"
+#include "sparse_blas/backends/mkl_common/mkl_handles.hpp"
+
+#include "oneapi/mkl/sparse_blas/detail/mklgpu/onemkl_sparse_blas_mklgpu.hpp"
+
+namespace oneapi::mkl::sparse::mklgpu {
+
+#include "sparse_blas/backends/mkl_common/mkl_spmm.cxx"
```

Comment:
Is there a reason you removed `mklgpu_operations.cpp` and separated it out into spmv/spmm/spsv files like this one? We could rename this file to `mkl_operations.cpp` and have:
```cpp
#include "sparse_blas/backends/mkl_common/mkl_spmm.cxx"
#include "sparse_blas/backends/mkl_common/mkl_spmv.cxx"
#include "sparse_blas/backends/mkl_common/mkl_spsv.cxx"
```

Reply:
There was no particular reasons, that's a good idea, done in https://github.com/oneapi-src/oneMKL/pull/500/commits/0bac3d49c56c48499e661d0cd4199407cbaabcc9

Reply:
Resoving old comment